pkg/apis/config/types.go

```diff
@@ -142,4 +158,11 @@ type NodeResourceTopologyMatchArgs struct {
 
 	// Namespaces to be considered by NodeResourceTopologyMatch plugin
 	Namespaces []string
+
+	// ScoringStrategy a scoring model that determine how the plugin will score the nodes.
+	ScoringStrategy *ScoringStrategy
```

Comment:
Thank you again for your review.
>If we just specify ScoringStrategyType here, we can use the concrete type (instead of pointer) in both internal and external types.go.

But now we also have the ```ResourceSpec[]``` [inside](https://github.com/kubernetes-sigs/scheduler-plugins/pull/220/commits/06ec7cd6189322449fb7f2e4edc643a84e5b8dd4) as well and not just the ```ScoringStrategyType```. this is why I kept it as a pointer.

Reply:
My 2 cents is that if a field is required, the definition in the internal version should be a concrete type. So I prefer to make it a concrete type, no matter it has a slice inside or not.

But I also noticed the upstream [version](https://github.com/kubernetes/kubernetes/blob/1f8b1b84fbfbf88bf50429b1f5673fd34ae30050/pkg/scheduler/apis/config/types_pluginargs.go#L93) also specified it as a pointer, so @alculquicondor could you confirm it's intended or not?

Reply:
It's not necessarily intended and we are not very consistent on this. Both would behave just the same from most users' perspective.

It just becomes extra cumbersome when it's a pointer to a native type such as `*int32`.

But +1 to concrete type for simplicity.